datasource/src/main/scala/quasar/blobstore/azure/AzureBlobstore.scala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2014–2018 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.blobstore.azure
+
+import slamdata.Predef._
+import quasar.api.resource.{ResourceName, ResourcePath, ResourcePathType}
+import quasar.blobstore.Blobstore
+import quasar.contrib.std.errorNotImplemented
+
+import java.lang.Integer
+import scala.collection.JavaConverters._
+
+import cats.effect._
+import cats.implicits._
+import com.microsoft.azure.storage.blob._
+import com.microsoft.azure.storage.blob.models._
+import com.microsoft.rest.v2.Context
+import fs2.Stream
+import io.reactivex._
+import io.reactivex.observers._
+
+object AzureBlobstore {
+
+  final class AsyncObserver[A] extends DisposableSingleObserver[A] {
+
+    @SuppressWarnings(Array("org.wartremover.warts.Null", "org.wartremover.warts.Var"))
+    private var callback: Either[Throwable, A] => Unit = _
+
+    def setCallback(cb: Either[Throwable, A] => Unit): Unit = {
+      this.callback = cb
+    }
+
+    override def onStart(): Unit = ()
+
+    override def onSuccess(a: A) =
+      if (callback != null) callback(Right(a))
+      else ()
+
+    override def onError(t: Throwable) =
+      if (callback != null) callback(Left(t))
+      else ()
+  }
+}
```

Comment:
This seems unnecessarily indirect. And… mutable. Why not just wrap the `single.subscribe` call in a `Async[F].async`?

Reply:
You mean how it was before https://github.com/rintcius/quasar-blobstore/blob/95908149a1ffeeb216a29824dd1624ce497704b8/datasource/src/main/scala/quasar/blobstore/azure/AzureBlobstore.scala ? That was not disposing the observer (which is part of the contract of `reactivex`). 
In a later commit the `dispose` was accounted for https://github.com/rintcius/quasar-blobstore/commit/ba872f400623d3bb23118b90a3f6443178fcf97d but it's a bit clumsy (and doesn't dispose in case of errors). 
So after discussing with @wemrysi I went for the current solution. Open for improvements though, I don't like the mutation either.